--- /etc/inc/notices.inc.orig    2022-04-29 19:52:54.937978000 -0300
+++ /etc/inc/notices.inc 2022-04-29 20:15:49.728255000 -0300
@@ -464,22 +464,28 @@
        }

        $url = "https://api.telegram.org/bot{$config['notifications']['telegram']['api']}/sendMessage?";
-       $data = array(
-               "chat_id" => $config['notifications']['telegram']['chatid'],
-               "text" => "{$config['system']['hostname']}.{$config['system']['domain']}\n{$message}"
-       );
-       $result = json_decode(curl_post_notification($url . http_build_query($data)), true);
-       if (is_array($result)) {
-               if ($result['ok']) {
-                       unset($err_msg);
+
+       $chat_ids = explode(":", $config['notifications']['telegram']['chatid']);
+       foreach($chat_ids as $chatid) {
+               $data = array(
+                       "chat_id" => $chatid,
+                       "text" => "{$config['system']['hostname']}.{$config['system']['domain']}\n{$message}"
+               );
+               $result = json_decode(curl_post_notification($url . http_build_query($data)), true);
+               if (is_array($result)) {
+                       if ($result['ok']) {
+                               unset($err_msg);
+                       } else {
+                               $err_msg = sprintf(gettext("Failed to send Telegram notification. Error received was :{$result['error_code']}: {$result['description']}"));
+                               log_error($err_msg);
+                       }
                } else {
-                       $err_msg = sprintf(gettext("Failed to send Telegram notification. Error received was :{$result['error_code']}: {$result['description']}"));
+                       $err_msg = gettext("API to Telegram did not return data in expected format!");
                        log_error($err_msg);
                }
-       } else {
-               $err_msg = gettext("API to Telegram did not return data in expected format!");
-               log_error($err_msg);
        }
+
+
        return $err_msg;
 }


--- /usr/local/pfSense/include/www/system_advanced_notifications.inc.orig      2022-04-29 20:27:05.154373000 -0300
+++ /usr/local/pfSense/include/www/system_advanced_notifications.inc   2022-04-29 18:48:02.213289000 -0300
@@ -243,7 +243,7 @@
                if (preg_replace("/[^a-zA-Z0-9_:\-]/", "", $config['notifications']['telegram']['api']) !== $config['notifications']['telegram']['api']) {
                        $input_errors[] = gettext("The only special characters permitted in the Telegram API string are _, - and :");
                }
-               if (preg_replace("/[^a-zA-Z0-9@_\-]/", "", $config['notifications']['telegram']['chatid']) !== $config['notifications']['telegram']['chatid']) {
+               if (preg_replace("/[^a-zA-Z0-9@_:\-]/", "", $config['notifications']['telegram']['chatid']) !== $config['notifications']['telegram']['chatid']) {
                        $input_errors[] = gettext("The Chat ID can only contain @, _ or - as special characters");
                }


--- /usr/local/www/system_advanced_notifications.php.orig      2022-04-30 00:12:30.183830000 +0000
+++ /usr/local/www/system_advanced_notifications.php   2022-04-30 00:12:40.395526000 +0000
@@ -234,7 +234,7 @@
        $pconfig['chatid'],
        ['placeholder' => '123456789']

-))->setHelp('Enter the chat ID number (private) or channel @username (public) that will be used to send the notifications to.');
+))->setHelp('Enter the chat ID number (private) or channel @username (public) that will be used to send the notifications to.<br/>Use colon (:) to separate multiple chat ids.');

 $section->addInput(new Form_Button(
        'test-telegram',
